### PR TITLE
Apply new widgets pattern to the Farm widget

### PR DESCRIPTION
**Pull request by codewiththeflowxyz**

### CODE REVIEW

**Review by inclinedmind — APPROVED** [review]
edit: oops wrong PR

File: apps/webapp/src/modules/farms/components/FarmFixedContent.tsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
           contractAddress={contractAddress}
+          connectButton={
+            <Button onClick={openConnectModal} data-testid="farm-widget-connect-wallet">
+              <Trans>Connect Wallet</Trans>
+            </Button>
+          }
           addRecentTransaction={addRecentTransaction}
-          connectButton={<div></div>}
+          locale={i18n.locale}
+          // onTransactionFinished={onTransactionFinished}
```
Comment (generatorpoint) [inline apps/webapp/src/modules/farms/components/FarmFixedContent.tsx:31]: Looking good so far. I'm working on a change that would remove the `connectButton` and lets us pass in the `openConnectModal` function as an `onConnect` callback. But if you finish this PR before me it won't be too hard to update later anyway.

Comment (codewiththeflowxyz) [reply apps/webapp/src/modules/farms/components/FarmFixedContent.tsx:31]: Great, let's do that.

**Review by bitsforpeace — APPROVED** [review]
Looking good. Noticed really minor thing, need some margin under the "Manage Position" header
Screenshot 2023-07-27 at 1 21 42 PM

**Review by iceanddust — APPROVED** [review]
Lgtm, just a couple of minor comments:

1. On the widget, I think it would be better if we could make this text span just 1 line instead of 2. Maybe changing it to `{stakeToken.symbol} amount` would do the trick?
image

2. This was not introduced in this PR, but it may be worth fixing it here as I don't think it'd involve a huge change: I noticed the navigation button for `Farms` in the top right corner redirects to a farm detail page instead of the farm list page, and also there's an extra character at the end of the URL
image
I think that comes from [this block of code](https://github.com/jetstreamgg/helipad/blob/main/apps/webapp/src/modules/layout/components/Header.tsx#L45-L51)

**Comment by generatorpoint** [comment]
fixed a small issue where we weren't showing the "no tokens staked" message
Screenshot 2023-07-27 at 3 41 09 PM

**Comment by generatorpoint** [comment]
> Lgtm, just a couple of minor comments:
> 
> 1. On the widget, I think it would be better if we could make this text span just 1 line instead of 2. Maybe changing it to `{stakeToken.symbol} amount` would do the trick?
>    image
> 2. This was not introduced in this PR, but it may be worth fixing it here as I don't think it'd involve a huge change: I noticed the navigation button for `Farms` in the top right corner redirects to a farm detail page instead of the farm list page, and also there's an extra character at the end of the URL
>    image
>    I think that comes from [this block of code](https://github.com/jetstreamgg/helipad/blob/main/apps/webapp/src/modules/layout/components/Header.tsx#L45-L51)

made these changes, except the farm detail redirect one. It looks intentional and I'm not sure what the desired behavior is.

**Review by inclinedmind — APPROVED** [review]

**Pull request merged**